security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/UserMappingOAuth2Service.scala

```diff
@@ -4,25 +4,41 @@ import io.circe.Decoder
 import pl.touk.nussknacker.ui.security.api.AuthenticatedUser
 import sttp.client3.SttpBackend
 
-import java.time.Instant
 import scala.concurrent.{ExecutionContext, Future}
 
 class UserMappingOAuth2Service[UserInfoData: Decoder, AuthorizationData <: OAuth2AuthorizationData: Decoder](
     delegate: OAuth2Service[UserInfoData, AuthorizationData],
-    loggedUserFunction: UserInfoData => AuthenticatedUser
+    loggedUserFunction: LoggedUserFunctionParameters[UserInfoData] => Future[AuthenticatedUser]
```

Comment:
WDYT about changing the param name to `authenticatedUserFrom`?

Reply:
Yeah I saw many readability problems around this code. One of the hardest for me was the inheritance structure: `OidcService <: GenericOidcService <: JwtOAuth2Service <: BaseOAuth2Service <: OAuth2Service`! Or methods like `obtainAuthorizationAndUserInfo` vs `checkAuthorizationAndObtainUserinfo`  I think that I could spend at least two days more making it more understandable but I didn't want to extend the scope of this task. But let's do some cleanups as you proposed - changed it

Reply:
I also changed all occurrences of `obtainUserInfo`. The need at the root is that we want to authenticate the user - obtaining the user info is just one step to do that. BTW this step is not always needed, sometimes all information is inside the access token or in id token so we don't obtain anything - only extract it.